### PR TITLE
deliveryservices/id/capacity no longer gives Internal Server Error 

### LINKED ISSUE #4237: deliveryservices/id/capacity gives Internal Server Error
## I'm submitting a ...

- [X] bug report
- [ ] new feature / enhancement request
- [ ] improvement request (usability, performance, tech debt, etc.)
- [ ] other 

## Traffic Control components affected ...

- [ ] CDN in a Box
- [ ] Documentation
- [ ] Grove
- [ ] Traffic Control Client
- [ ] Traffic Monitor
- [X] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [ ] unknown

## Current behavior:

When making a `GET` request to `/deliveryservices/{{id}}/capacity` where the deliverservice id pertains to a delivery service that was recently created and has no server assignments, the response returned is: 
```
HTTP/1.1 500 Internal Server Error
{
    "alerts": [
        {
            "text": "Internal Server Error",
            "level": "error"
        }
    ]
}
```
## Expected / new behavior:

If there is no capacity that can be obtained it should return:
```
HTTP/1.1 200 OK
{
    "response": {
        "availablePercent": 0,
        "unavailablePercent": 0,
        "utilizedPercent": 0,
        "maintenancePercent": 0
    }
}
```

## Anything else:

`error.log` shows: `getting delivery service capacity: capacity was zero!`

### PR DESCRIPTION
**Pull request by rimashah25**
## What does this PR (Pull Request) do?

- [x] This PR fixes #4237 

## Which Traffic Control components are affected by this PR?

- Traffic Ops

## What is the best way to verify this PR?

1. Run CIAB
2. Open Traffic Portal, create a delivery service (ds).
3. Run https://TP-url:port/api/3.0/deliveryservices/ds-id/capacity
Note: You should see the following response:
```
{"response":
    {
     "availablePercent":0,
     "unavailablePercent":0,
     "utilizedPercent":0,
     "maintenancePercent":0
    }
}
```

## If this is a bug fix, what versions of Traffic Control are affected?

- master 00294da

## The following criteria are ALL met by this PR

- [x] This PR includes tests
- [x] I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information

Documentation has a correct example.

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/deliveryservice/capacity.go [hunk 1]
```diff
@@ -135,6 +139,20 @@ func getCapacity(tx *sql.Tx, ds tc.DeliveryServiceName, cdn tc.CDNName) (Capacit
 	}, nil
 }
 
+// To check whether a given DS has servers.
+func dsHasSever(ds tc.DeliveryServiceName, crConfig tc.CRConfig) bool {
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/capacity.go:143]: Nitpick: typo, `dsHasServer`

File: traffic_ops/traffic_ops_golang/deliveryservice/capacity.go [hunk 2]
```diff
@@ -135,6 +139,20 @@ func getCapacity(tx *sql.Tx, ds tc.DeliveryServiceName, cdn tc.CDNName) (Capacit
 	}, nil
 }
 
+// To check whether a given DS has servers.
```
Comment (rob05c) [inline traffic_ops/traffic_ops_golang/deliveryservice/capacity.go:142]: Nitpick: GoDoc comments need to be complete sentences starting with the symbol name. Should be
```
 // dsHasServer checks whether a given DS has servers. 
```

**Review by rob05c — APPROVED** [review]
Looks good! Code looks good. I manually tested, behaves as expected, DSes with no assigned servers return 0 instead of 500. API tests pass, unit tests pass.

The nitpicks below would be nice, but I wouldn't hold up merging this for them.

File: traffic_ops/traffic_ops_golang/deliveryservice/capacity.go (hunk 1 above)
Comment (rimashah25) [reply traffic_ops/traffic_ops_golang/deliveryservice/capacity.go:143]: Updated.

File: traffic_ops/traffic_ops_golang/deliveryservice/capacity.go (hunk 2 above)
Comment (rimashah25) [reply traffic_ops/traffic_ops_golang/deliveryservice/capacity.go:142]: Updated.

**Pull request merged**